test/gtest/ucs/test_async.cc

```diff
@@ -537,8 +537,12 @@ class local_timer_remove_handler : public local_timer {
 
 UCS_TEST_P(test_async, timer_unset_from_handler) {
     local_timer_remove_handler lt(GetParam());
-    suspend_and_poll(&lt, COUNT);
+    ucs_time_t deadline = ucs_get_time() + ucs_time_from_sec(10.0);
+    do {
+        suspend_and_poll(&lt, 1);
+    } while ((lt.count() == 0) && (ucs_get_time() < deadline));
     EXPECT_GE(lt.count(), 1);
```

Comment:
is it ok that timeout is not processed? (will fail by assert)

Reply:
not sure what you mean.. but in case of timeout the next EXPECT_EQ() should fail

Reply:
right, my question was is it expected behavior? :)
ok, got it